### PR TITLE
Add status code 422 as book visit response,

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/config/OrchestrationExceptionHandler.kt [hunk 1]
```diff
@@ -126,14 +127,20 @@ data class ErrorResponse(
...
     errorCode: Int? = null,
     userMessage: String? = null,
     developerMessage: String? = null,
-    moreInfo: String? = null,
   ) :
-    this(status.value(), errorCode, userMessage, developerMessage, moreInfo)
+    this(status.value(), errorCode, userMessage, developerMessage)
 }
+
+data class ApplicationValidationErrorResponse(
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/config/OrchestrationExceptionHandler.kt:140]: Do we want to use inheritance here as well? To keep consistent with the other changes made

data class ApplicationValidationErrorResponse(
  val validationErrors: List<ApplicationValidationErrorCodes>,
) : ErrorResponse(status = HttpStatus.UNPROCESSABLE_ENTITY)

**Review by adaviesMOJ — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt [hunk 2]
```diff
@@ -173,7 +177,28 @@ class VisitSchedulerClient(
...
+      }.block(apiTimeout)
+  }
+
+  private fun getBookVisitApplicationValidationErrorResponse(e: Throwable): Throwable {
+    if (e is WebClientResponseException && isUnprocessableEntityError(e)) {
+      try {
+        val errorResponse = objectMapper.readValue(e.responseBodyAsString, ApplicationValidationErrorResponse::class.java)
+        return ApplicationValidationException(errorResponse.validationErrors)
+      } catch (jsonProcessingException: Exception) {
+        LOG.error("An error occurred processing the application validation error response - ${e.stackTraceToString()}")
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt:197]: weirdly i think this needs an exception haha?

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt:197]: updated to throw an Exception instead.

Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt:197]: sounds good!

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**